bundles/GraphicServiceOrder/Service/OrderService.php

```diff
@@ -47,16 +53,65 @@ public function __construct(
         GsOrderRepository $gsOrderRepository,
         KernelInterface $appKernel,
         MessageBusInterface $messageBus,
-        string $ownCloudFilesFolder
+        string $ownCloudFilesFolder,
+        TokenStorageInterface $tokenStorage,
+        FileUploader $fileUploader,
+        UserManager $userManager
     ) {
         $this->entityManager = $entityManager;
-        $this->gsOrderRepository = $gsOrderRepository;
         $this->hammerService = $hammerService;
         $this->ownCloudService = $ownCloudService;
         $this->gsOrderRepository = $gsOrderRepository;
         $this->appKernel = $appKernel;
         $this->messageBus = $messageBus;
         $this->ownCloudFilesFolder = $ownCloudFilesFolder;
+        $this->tokenStorage = $tokenStorage;
+        $this->fileUploader = $fileUploader;
+        $this->userManager = $userManager;
+    }
+
+    /**
+     * Preset some values from user entity.
+     *
+     * @return \GraphicServiceOrder\Entity\GsOrder
+     */
+    public function prepareOrder()
+    {
+        $gsOrder = new GsOrder();
+        $token = $this->tokenStorage->getToken();
+        if (null !== $token) {
+            $user = $token->getUser();
+            $gsOrder
+                ->setFullName($user->getFullName())
+                ->setAddress($user->getAddress())
+                ->setDepartment($user->getDepartment())
+                ->setPostalcode($user->getPostalCode())
+                ->setCity($user->getCity());
+        }
+
+        return $gsOrder;
+    }
+
+    /**
+     * Update active user with submitted values.
+     *
+     * @param $gsOrder
+     */
+    private function updateUser($gsOrder)
```

Comment:
Would probably change the function name to something like
updateUserWithGSOrder($gsOrder)
instead

Reply:
changed